sdk/attestation/azure-security-attestation/azure/security/attestation/_administration_client.py

```diff
@@ -0,0 +1,212 @@
+        if self._config.token_validation_options.validate_token:
+            if not token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs)):
+                raise Exception("Token Validation of PolicySet API failed.")
+
+
+        return AttestationResponse[PolicyResult](token, token.get_body())
+
+    @distributed_trace
+    def get_policy_management_certificates(self, **kwargs):
+        #type:(Any) -> AttestationResponse[list[list[bytes]]]
+        """ Retrieves the set of policy management certificates for the instance.
+
+        The list of policy management certificates will only be non-empty if the
+        attestation service instance is in Isolated mode.
+
+        :return AttestationResponse[list[list[bytes]]: Attestation service response 
+            encapsulating a list of DER encoded X.509 certificate chains.
+        """
+
+        cert_response = self._client.policy_certificates.get(**kwargs)
+        token = AttestationToken[PolicyCertificatesResult](
+            token=cert_response.token,
+            body_type=PolicyCertificatesResult)
+        if self._config.token_validation_options.validate_token:
+            if not token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs)):
+                raise Exception("Token Validation of PolicyCertificates API failed.")
+        certificates = list()
+
+        cert_list = token.get_body()
+
+        for key in cert_list.policy_certificates.keys:
+            key_certs = list()
+            for cert in key.x5_c:
+                key_certs.append(base64.b64decode(cert))
+            certificates.append(key_certs)
+        return AttestationResponse[list](token, certificates)
+
+    @distributed_trace
+    def add_policy_management_certificate(self, certificate_to_add, signing_key, **kwargs):
+        #type:(bytes, AttestationSigningKey, Any)-> AttestationResponse[PolicyCertificatesModificationResult]
+        key=JSONWebKey(kty='RSA', x5_c = [ base64.b64encode(certificate_to_add).decode('ascii')])
+        add_body = AttestationCertificateManagementBody(policy_certificate=key)
+        cert_add_token = AttestationToken[AttestationCertificateManagementBody](
+            body=add_body,
+            signer=signing_key,
+            body_type=AttestationCertificateManagementBody)
+
+        cert_response = self._client.policy_certificates.add(cert_add_token.serialize(), **kwargs)
+        token = AttestationToken[PolicyCertificatesModificationResult](token=cert_response.token,
+            body_type=PolicyCertificatesModificationResult)
+        if self._config.token_validation_options.validate_token:
+            if not token.validate_token(self._config.token_validation_options, self._get_signers(**kwargs)):
+                raise Exception("Token Validation of PolicyCertificate Add API failed.")
+        return AttestationResponse[PolicyCertificatesModificationResult](token, token.get_body())
```

Comment:
Knowing that the token doesn't always come from the `get_body()` method, I think this is a fine implementation and we can work on something simpler for a later preview. This is all in the convenience wrapper so there should be no difference to the end user.

Reply:
Thanks. I started working through this for a while and realized that because I was hiding the generated types, the situation where get_body() wasn't used was becoming increasingly rare, especially since I'm going to be working on hiding the generated types completely in the convenience layer.